docs/source/deprecation.md

```diff
@@ -1,17 +1,458 @@
+
+def produce_answer():
+    api.produce(Foo(value='Answer is: 42'))
+```
+
+And the updated actor looks like this:
+
+```python
+from leapp.actors import Actor
+from leapp.libraries.actor import fooproducer_lib
+from leapp.models import Foo  # deprecated model
+from leapp.tags import IPUWorkflowTag, FactsPhaseTag
+from leapp.utils.deprecation import suppress_deprecation
+
+
+@suppress_deprecation(Foo)
+class FooProducer(Actor):
+    """
+    Just produce the right answer to the world.
+    """
+
+    name = 'foo_producer'
+    consumes = ()
+    produces = (Foo,)
+    tags = (IPUWorkflowTag, FactsPhaseTag)
+
+    def process(self):
+        fooproducer_lib.produce_answer()
+```
+
+Now, if you execute the actor again - you can see that there is still not any
+deprecation message. So the `suppress_deprecation` decorator works transitively
+as expected. However, even when the actor is treated well, the current
+implementation could affect results of unit-tests. Just to show the idea, what
+could be wrong, imagine unit test like this one (ignore the meaninfulness
+of this example test, it's just for education):
+
+```python
+from leapp.libraries.actor import fooproducer_lib
+from leapp.libraries.common.testutils import produce_mocked
+from leapp.libraries.stdlib import api
+from leapp.models import Foo  # deprecated model
+
+def test_process(monkeypatch):
+    produced_msgs = produce_mocked()
+    monkeypatch.setattr(api, 'produce', produced_msgs)
+    fooproducer_lib.produce_answer()
+    assert Foo(value='Answer is: 42') in produced_msgs.model_instance
+```
 
-### Warning message display behaviour
+If you run the test, you will get output like this (shortened):
+
+```python
+| 21:48:01 | conftest | INFO | conftest.py | Actor 'foo_producer' context teardown complete
+
+repos/system_upgrade/el7toel8/actors/fooproducer/tests/test_unit_fooproducer.py::test_process PASSED
+
+================================================== warnings summary ==================================================
+repos/system_upgrade/el7toel8/actors/fooproducer/tests/test_unit_fooproducer.py::test_process
+  /tmp/leapp-repository/repos/system_upgrade/el7toel8/actors/fooproducer/libraries/fooproducer_lib.py:5: _DeprecationWarningContext: Usage of deprecated Model "Foo"
+    api.produce(Foo(value='Answer is: 42'))
+  /tmp/leapp-repository/repos/system_upgrade/el7toel8/actors/fooproducer/tests/test_unit_fooproducer.py:10: _DeprecationWarningContext: Usage of deprecated Model "Foo"
+    assert Foo(value='Answer is: 42') in produced_msgs.model_instances
+
+-- Docs: http://doc.pytest.org/en/latest/warnings.html
+======================================== 1 passed, 2 warnings in 0.13 seconds ========================================
+```
+
+As you can see the warning have been generated again. This time on two places,
+in `test_process` and `produce_answer` functions. Unless warning messages affect
+results of tests, we do not require strictly to handle them. However it's good
```

Comment:
I think it's not good idea. We already discussed something like that in the past and we agreed in the end just on the soft way for custom/third party code. In such case, we are not talking about the deprecation. It should be there until we remove it. With no changed functionality. Such blockers do not make sense really. Maybe it could result in state when people stop writting tests.